### PR TITLE
Fix crash on Windows in focal length collection filter

### LINKED ISSUE #11737: [filtering | crash] darktable with new collection filters crashes on Windows in focal length filter
**Describe the bug/issue**

Platform-dependent (only Windows, not happens with Linux build, not tested on mac OS) crash while trying to use focal length filter from new collection filters module.

**To Reproduce**
1. Build latest master on Windows
2. Open darktable
3. Add new filtering rule that works on focal length
4. Hover your mouse over widget of this new rule
5. darktable will crash immediately without even producing backtrace file

**Expected behavior**

darktable should work as expected without crashing.

**Which commit introduced the error**

Probably this bug existed from the first commits that introduced new collection filters. At least I had the opportunity to check it out on the build 3.9.0+1228 and this new filters feature was not changed too much between merging and commit of that build.

**Platform**
* darktable version : from at least 3.9.0+1228 to current master, 3.9.0+1407
* OS  : Windows 10 21H2 (OS Build 19044.1645)
* OpenCL : should not be relevant as filtering code clearly does not use it (and yes, tested that darktable crashes with --disable-opencl)

* CMAKE_BUILD_TYPE  :  RelWithDebInfo

**Additional context**

The crash is not data dependent as it also appears when used an empty/new config-dir.
I did not use Lua scripts.

### PR DESCRIPTION
**Pull request by victoryforce**
Fixes #11737.

### CODE REVIEW

File: src/libs/filters/focal.c [hunk 1]
```diff
@@ -71,7 +71,7 @@ static gchar *_focal_print_func(const double value, const gboolean detailled)
 
   if(detailled)
   {
-    dt_util_dstrcat(txt, " %s", _("mm"));
+    return dt_util_dstrcat(txt, " %s", _("mm"));
```
Comment (TurboGit) [inline src/libs/filters/focal.c:74]: or probably more readable:
`txt = dt_util_dstrcat(txt, " %s", _("mm"));`

**Review by TurboGit — APPROVED** [review]
Thanks!

File: src/libs/filters/focal.c (hunk 1 above)
Comment (AlicVB) [reply src/libs/filters/focal.c:74]: good catch... I wonder how it could even work under linux...
thanks for the fix !

Comment (victoryforce) [reply src/libs/filters/focal.c:74]: For me both versions equally readable. And version with assignment even looks not so **logical** (and therefore less understandable). A question would arise in my head: what is purpose of unneeded assigning to variable when we will immidiately end execution flow of function and return this value (saying that I believe that modern compilers optimize this to the same code).

I just looked at how it is done in similar case in another filter (ratio.c) and did similarly.

**Pull request merged**